ui/apps/dashboard/src/app/(organization-active)/(dashboard)/env/[environmentSlug]/manage/[ingestKeys]/[keyID]/page.tsx

```diff
@@ -112,18 +118,25 @@ export default function Keys({ params: { ingestKeys, keyID } }: KeyDetailsProps)
                 </DropdownMenuItem>
               </DropdownMenuContent>
             </DropdownMenu>
-
             <DeleteKeyModal
               keyID={keyID}
               isOpen={isDeleteKeyModalVisible}
               onClose={() => setIsDeleteKeyModalVisible(false)}
+              description={
+                isIntegration
+                  ? 'Warning: This key was created via integration. Please confirm that you are no longer using it before deleting.'
```

Comment:
This being a warning it might make sense to use the `<Alert>` component in the future, but the best would be to make the AlertModal more composable, so it can be for later

Reply:
Good idea! I was thinking about doing this, but avoided the refactor as `description` was a `string` type. I think all of our confirmation modals and dialogs could use some cleanup and a new design with options.